onnx/funconnx/op_run.py

```diff
@@ -0,0 +1,508 @@
+            if linked_attributes is None:
+                raise AttributeError(
+                    f"One attribute is linked but no linked value is provided, "
+                    f"in class {type(self)}."
+                )
+            for att in self.attributes_names_:
+                v = getattr(self, att)
+                if isinstance(v, RefAttrName):
+                    if v.name not in linked_attributes:
+                        raise ValueError(
+                            f"Unable to find a value for linked attribute {att!r} in {linked_attributes!r} "
+                            f"in node {type(self)}."
+                        )
+                    overridden_attributes[att] = linked_attributes[v.name]
+
+        self._log("-- begin %s.run(%d inputs)", self.__class__.__name__, len(args))
+        kwargs = kwargs.copy()
+        for att in self.attributes_names_:
+            if att in overridden_attributes or att in kwargs:
+                continue
+            if not hasattr(self, att):
+                raise NameError(
+                    f"Attribute {att!r} is missing in operator {self.__class__.__name__!r}."
+                )
+            kwargs[att] = getattr(self, att)
+        try:
+            if len(overridden_attributes) > 0:
+                res = self._run(*args, **overridden_attributes, **kwargs)
+            else:
+                res = self._run(*args, **kwargs)
+        except (TypeError, AttributeError) as e:
+            raise TypeError(
+                f"Issues with types {[type(_) for _ in args]} and attributes "
+                f"{list(sorted(kwargs))} and linked attributes={list(sorted(overridden_attributes))} "
+                f"(operator {self.__class__.__name__!r})."
+            ) from e
+        self._log("-- done %s.run -> %d outputs", self.__class__.__name__, len(res))
+        if not isinstance(res, tuple):
+            raise TypeError(
+                f"Method '_run' of class {self.__class__.__name__!r} does not return a tuple but {type(res)}."
+            )
+        if len(res) == 0:
+            raise ValueError(
+                f"Method '_run' of class {self.__class__.__name__!r} does not return any result."
+            )
+        if any(map(lambda t: isinstance(t, tuple), res)):
+            dtypes = [type(t) for t in res]
+            raise TypeError(
+                f"Method '_run' of class {self.__class__.__name__!r} must not return a tuple: {dtypes}."
+            )
+        return res
+
+    @classmethod
+    def make_node(
+        cls,
+        n_inputs: Optional[int] = None,
+        n_outputs: Optional[int] = None,
+        **kwargs: Any,
+    ) -> NodeProto:  # type: ignore
+        """
+        Creates an ONNX node for this class based on the given information.
+
+        :param n_inputs: number of inputs (default is defined by the operator schema)
+        :param n_outputs: number of outputs (default is defined by the operator schema)
+        :param verbose: verbosity
```

Comment:
Seems to be missing from parameter list above. Is it supposed to be part of kwargs?

Reply:
I guess a copy-paste error?

Reply:
The operator name and domain is already defined by class cls (the runtime). It can only execute a specific node. So it is not needed.

Reply:
I don't understand. I meant that the description about the parameter `verbose` seems like it is here by mistake?